readthedocs/projects/views/private.py

```diff
@@ -143,25 +150,36 @@ def get_success_url(self):
         return reverse('projects_detail', args=[self.object.slug])
 
 
-@login_required
-def project_version_detail(request, project_slug, version_slug):
-    """Project version detail page."""
-    project = get_object_or_404(
-        Project.objects.for_admin_user(request.user),
-        slug=project_slug,
-    )
-    version = get_object_or_404(
-        Version.internal.public(
-            user=request.user,
-            project=project,
+class ProjectVersionMixin(ProjectAdminMixin, PrivateViewMixin):
+
+    model = Version
+    context_object_name = 'version'
+    form_class = VersionForm
+    lookup_url_kwarg = 'version_slug'
+    lookup_field = 'slug'
+
+    def get_success_url(self):
+        return reverse(
+            'project_version_list',
+            kwargs={'project_slug': self.get_project().slug},
+        )
+
+
+class ProjectVersionDetail(ProjectVersionMixin, UpdateView):
```

Comment:
This should be refactored to be 2 views: detail and update. I can create a separate issue for that.